test/unit/cli/test_cli.py

```diff
@@ -0,0 +1,145 @@
+    ],
+)
+def test_missing_scs_file(args):
+    result = CliRunner().invoke(*args)
+    assert_error(result, "Error: Missing argument 'SCS_FILE'")
+
+
+@pytest.fixture
+def scs_file(tmp_path) -> Path:
+    return tmp_path / "sample.sqlite"
+
+
+@pytest.fixture
+def sample_scs(scs_file, monkeypatch) -> Secrets:
+    password = "sample password"
+    monkeypatch.setitem(os.environ, "SCS_MASTER_PASSWORD", password)
+    return Secrets(scs_file, password)
+
+
+@pytest.mark.parametrize(
+    "backend, expected",
+    [
+        ("onprem", "on-premise"),
+        ("saas", "SaaS"),
+        ("docker-db", "Docker"),
+    ],
+)
+def test_configure(backend, expected, sample_scs):
+    result = CliRunner().invoke(
+        commands.configure,
+        [backend, str(sample_scs.db_file)],
+    )
+    assert result.exit_code == 0
+    assert BackendSelector(sample_scs).backend_name == expected
+
+
+@pytest.mark.parametrize(
+    "command, kwargs, secrets, expected_show",
+    [
+        (
+            "docker-db",
+            {},
+            {},
+            """
+            backend: onprem
+            use_itde: True
+            --db-mem-size: 2
+            --db-disk-size: 2
+            --accelerator: none
+            --db-schema: SSS
+            """,
+        ),
+        (
+            "onprem",
+            {
+                "--db-username": "UUU",
+            },
+            {
+                "--db-password": "SCS_EXASOL_DB_PASSWORD",
+                "--bucketfs-password": "SCS_BUCKETFS_PASSWORD",
+            },
+            """
+            backend: onprem
+            use_itde: False
+            --db-host-name: localhost
+            --db-port: 8563
+            --db-username: UUU
+            --db-password: ****
+            --db-use-encryption: True
+            --bucketfs-host: localhost
+            --bucketfs-host-internal: localhost
+            --bucketfs-port: 2580
+            --bucketfs-port-internal: 2580
+            --bucketfs-user: w
+            --bucketfs-password: ****
+            --bucketfs-name: bfsdefault
+            --bucket: default
+            --bucketfs-use-encryption: True
+            --ssl-use-cert-validation: True
+            --db-schema: SSS
+            """,
+        ),
+        (
+            "saas",
+            {
+                "--saas-database-name": "DB",
+                "--saas-account-id": "acc",
+                "--saas-url": "URL",
+            },
+            {"--saas-token": "SCS_EXASOL_SAAS_TOKEN"},
+            """
+            backend: saas
+            use_itde: False
+            --saas-url: URL
+            --saas-account-id: acc
+            --saas-database-name: DB
+            --saas-token: ****
+            --ssl-use-cert-validation: True
+            --db-schema: SSS
+            """,
+        ),
+    ],
+)
+def test_round_trip(command, kwargs, secrets, expected_show, monkeypatch, sample_scs):
+    def cmd_args():
+        yield from [command, scs_file, "--db-schema", "SSS"]
```

Comment:
Why the "--db-schema" is hardcoded here?

Reply:
Because it's identical to all the different backends. 
If you prefer, then we can also move it to the individual entries of the parameterized test.